test/Npgsql.Tests/Types/EnumTests.cs

```diff
@@ -117,6 +118,24 @@ public async Task Unmapped_enum_as_string()
         await AssertType(connection, "happy", "happy", type, npgsqlDbType: null, isDefaultForWriting: false);
     }
 
+    [Test]
+    public async Task Unmapped_unknown_enum_as_int()
+    {
+        await using var connection = await OpenConnectionAsync();
+
+        await AssertTypeWrite(UnmappedEnum.Happy, "2", "integer", NpgsqlDbType.Integer, DbType.Int32, isDefault: false, isNpgsqlDbTypeInferredFromClrType: false);
```

Comment:
Maybe also add testing to verify that reading is not supported (there's AssertTypeUnsupportedRead)